app/src/main/java/com/github/se/cyrcle/model/zone/Zone.kt

```diff
@@ -20,7 +25,101 @@ data class Zone(
   constructor(
       boundingBox: BoundingBox,
       name: String
-  ) : this(boundingBox, name, LocalTime.now(), generateZoneUid())
+  ) : this(boundingBox, name, LocalDate.now(), generateZoneUid())
+
+  /** Returns a JSON representation of the zone. */
+  fun toJson(): String {
```

Comment:
I feel like this function needs a lot of maintenance. As soon as you add a field you have to modify the toJson and fromJson functions. Wouldn't it be better to proceed like in ParkingRepositoryFirestore and have an adapter for the Bounding Box. Have a look at `BoundingBoxTypeAdapter()` provided by mapbox first to make sure it doesn't work how you'd like it to before looking into making a custom one.

If you prefer keeping a man made serializer, maybe you could use a json Builder as I really feel like this method is error prone

Reply:
@Armetiis , You're right it's not pretty.
I fixed it take a look at the corresponding commit
I had to make a customTypeAdapter for the localDate, and learn how to use Gson, but I think it looks cleaner now.